src/main/java/rx/internal/operators/OperatorMerge.java

```diff
@@ -847,8 +847,11 @@ public void onNext(T t) {
         }
         @Override
         public void onError(Throwable e) {
-            done = true;
+            // Need to queue the error first before setting done, so that after emitLoop() removes the subscriber,
+            // it is guaranteed to notice the error. Otherwise it would be possible that inner subscribers count was 0,
+            // and at the same time the error queue was empty.
             parent.getOrCreateErrorQueue().offer(e);
+            done = true;
```

Comment:
Please add an unit test that checks this race condition.

Reply:
How do I do this? This looks like a very rare / hard to trigger one. A thread would have to stop for a moment after setting done and before queuing the error at the time when emitLoop was running on another thread.

Reply:
That's what `TestUtil.race` is for. Have one subject `onCompleted()` and another `onError()`, `mergeDelayError` them. Loop this at least 1000 times, check a `TestSubscriber` for error termination. Remember to create the subjects and subscriber inside the loop due to the terminal event nature.